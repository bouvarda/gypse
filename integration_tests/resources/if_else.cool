
class Main() {
    {
        var a: Int = 1;
        if (a >= 0) {
          out_string("positive");
        } else {
          out_string("negative");
        };

        out_nl();
        var b: Int = 116;
        if (b < 100) {
            out_string("low");
        } else {
            out_string("high");
        };

        out_nl();
        if (a == b) {
            out_string("No/");
        } else {
            out_string("Yes/");
        };
        if (a != b) {
            out_string("Yes/");
        } else {
            out_string("No/");
        };
        if (a > b) {
            out_string("No/");
        } else {
            out_string("Yes/");
        };
        if (a < b) {
            out_string("Yes/");
        } else {
            out_string("No/");
        };

        a = a + 1;
        if (a == 0) {
            out_string("No");
        } else if (a == 1) {
            out_string("No");
        } else if (a == 2) {
            out_string("Yes");
        } else if (a == 3) {
            out_string("No");
        } else {
            out_string("No");
        };
    };
}