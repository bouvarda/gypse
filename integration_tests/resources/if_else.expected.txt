positive
high
Yes/Yes/Yes/Yes/Yes